src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -1086,6 +1087,7 @@ private void WriteString(in DbRow row, Utf8JsonWriter writer)
             database.CompleteAllocations();
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
How about instead moving the throw into its own ThrowHelper and letting the inliner do its normal thing for the check code?

Reply:
in fact one already exists:
https://github.com/dotnet/runtime/blob/f2a86839180e50d6edb12e0a02875528a7e01836/src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs#L674

Reply:
Good idea, addressed & validated benchmarks again - they still show improvements:
```

|         Method |        Job |                   Toolchain |     Mean |     Error |    StdDev | Ratio |
|--------------- |----------- |---------------------------- |---------:|----------:|----------:|------:|
| EnumerateArray | Job-XYJNOR |      \Core_Root\corerun.exe | 1.090 us | 0.0033 us | 0.0027 us |  1.00 |
| EnumerateArray | Job-IXOYGD | \Core_Root_base\corerun.exe | 1.311 us | 0.0060 us | 0.0053 us |  1.20 |
```